python/shotgun_fields/status_list_widget.py

```diff
@@ -41,3 +37,63 @@ def _string_value(self, value):
             str_val = ("<span style='color: rgb(%s)'>&#9608;</span>&nbsp;%s" % (color_str, str_val))
 
         return str_val
+
+
+class StatusListEditorWidget(QtGui.QComboBox):
+    """
+    Allows editing of a ``status_list`` field value as returned by the Shotgun API.
+    """
+    __metaclass__ = ShotgunFieldMeta
+    _EDITOR_TYPE = "status_list"
+    _IMMEDIATE_APPLY = True
+
+    def get_value(self):
+        """
+        :return: The internal value being displayed by the widget.
+        """
+        return self.itemData(self.currentIndex())
+
+    def setup_widget(self):
+        """
+        Prepare the widget for display
+
+        Called by the metaclass during initialization.
+        """
+        self.addItem("")
+        self.setMinimumWidth(100)
+
+        valid_values = shotgun_globals.get_valid_values(self._entity_type, self._field_name)
+        for value in valid_values:
+            self.addItem(shotgun_globals.get_status_display_name(value), value)
+
+        self.activated.connect(
+            lambda i: self.value_changed.emit()
+        )
+
+    def _begin_edit(self):
+        """
+        Prepare the widget for editing by showing the popup
+        """
+        self.adjustSize()
+        self.showPopup()
+
+    def _display_default(self):
+        """
+        Display the default value of the widget.
+        """
+        self.setCurrentIndex(0)
+
+    def _display_value(self, value):
+        """
+        Set the value displayed by the widget.
+
+        :param value: The value returned by the Shotgun API to be displayed
+        """
+        if value is None:
+            self.setCurrentIndex(0)
```

Comment:
Yep. and tweaked the code below in case the value couldn't be found to set to 0 as well.